python/pyspark/sql/tests.py

```diff
@@ -476,6 +476,19 @@ def test_udf_defers_judf_initalization(self):
             "judf should be initialized after UDF has been called."
         )
 
+    @unittest.skipIf(sys.version_info < (3, 3), "Unittest < 3.3 doesn't support mocking")
```

Comment:
You can create a testcase without the Spark base and verify that creating a UDF doesn't create a SparkContext. This does not require making a SparkContext.